addons/web/static/src/core/code_editor/code_editor.js

```diff
@@ -177,4 +192,22 @@ export class CodeEditor extends Component {
 
         onResized(this.editorRef, debouncedResize);
     }
+
+    async highlightInvalidElements(arch) {
+        const invalidXpathElems = await this.orm.call("ir.ui.view", "get_invalid_xpath_elements", [this.resId, arch]);
+        const xpathPatterns = invalidXpathElems.map((elem) => escapeRegExp(elem)).join('|');
+
+        const xpathRegex = new RegExp(`<xpath\\s+([^>]*?)expr\\s*=\\s*"(${xpathPatterns})"[^>]*>`, 'g');
+        const { doc }  = this.aceEditor.session;  
```

Comment:
- there are two spaces before `=` => only one needed.
- there are two spaces at the end of the line => remove them.